examples/ccn-lite-relay/Makefile

```diff
@@ -38,8 +38,9 @@ USEMODULE += xtimer
 USEMODULE += random
 USEMODULE += prng_xorshift
 
-USEPKG += tlsf
+USEMODULE += tlsf-malloc
 
+USEPKG += tlsf
```

Comment:
You could model `tlsf` as a dependency for `tlsf-malloc`. Using `tlsf-malloc` without `tlsf` doesn't make much sense, or am I missing something? This way, we could skip the explicit `USEPKG += tlsf` statement.

--

Just saw that you wrote `USEPKG += tlsf` in `pkg/tlsf/contrib/Makefile.dep`, maybe that's eough? Can we drop the explicit `USEPKG += tlsf`?

Reply:
That's what I tried to do. However, the +DIR is in tlsf's Makefile.include and I think it doesn't get read if the USEPKG is not there. It's tricky because the contrib is a module that lives inside a package, I'm not sure if it can be done "the right way".

Reply:
@jcarrano I would rather have this line removed. We can achieve that by setting a dependency in `Makefile.dep`:

```make
ifneq (,$(filter tlsf-malloc,$(USEMODULE)))
  USEPKG += tlsf
endif
```
This resolves the dependencies in the correct order.

Reply:
That does not work. The Makefiles in the `tlsf` directory only get read if `USEPKG += tlsf`. In the other packages that have contrib directorties, the contib gets added to `USEMODULE` **always** if the parent package is used.
It is a limitation of the current build system: if a package has an optional submodule, then the package **and** the module must be explicitly selected.

Reply:
But when I apply my proposed code snipped, then it still builds and executes for me..

Reply:
You have to put it around line 726, beore the call to
```make
-include $(USEPKG:%=$(RIOTPKG)/%/Makefile.dep)
```
where the `USEPKG` statements are resolved

Reply:
In case I wasn't clear enough: I was talking about the main `Makefile.dep` in the RIOT root

Reply:
Ok, I didn't know that file existed. I wish I could keep all of TLSF inside pkg/tlsf without having to touch global files, but I'll do it anyways.

Reply:
>I wish I could keep all of TLSF inside pkg/tlsf without having to touch global files, but I'll do it anyways.

I understand your motivation, but it looks nicer and is less error prone if you only have to use `USEMODULE += tlsf-malloc` in the user application instead of also specifying the package (:
And we already have this global depencendy file .. so let's make use of it.